nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/hosted/ConanHostedMetadataFacetSupport.java

```diff
@@ -0,0 +1,131 @@
+import org.sonatype.repository.conan.internal.metadata.ConanCoords;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.google.common.hash.HashCode;
+import org.apache.commons.lang3.tuple.Pair;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static java.util.stream.Collectors.toMap;
+import static org.sonatype.repository.conan.internal.utils.ConanFacetUtils.findAsset;
+
+/**
+ * @since 1.0.0
+ */
+public class ConanHostedMetadataFacetSupport
+    extends FacetSupport
+{
+  private static final List<AssetKind> DOWNLOAD_URL_ASSET_KINDS = Collections
+      .unmodifiableList(Arrays.asList(
+          AssetKind.CONAN_EXPORT,
+          AssetKind.CONAN_FILE,
+          AssetKind.CONAN_SOURCES,
+          AssetKind.CONAN_MANIFEST)
+      );
+
+  private static final List<AssetKind> DOWNLOAD_URL_PACKAGE_ASSET_KINDS = Collections
+      .unmodifiableList(Arrays.asList(
+          AssetKind.CONAN_PACKAGE,
+          AssetKind.CONAN_INFO,
+          AssetKind.CONAN_MANIFEST)
+      );
+
+  private static Map<String, String> generateDownloadUrls(
+      final List<AssetKind> assetKinds,
+      final ConanCoords coords, final String repositoryUrl)
+  {
+    return assetKinds
+        .stream()
+        .collect(toMap(AssetKind::getFilename,
+            x -> repositoryUrl + "/" + ConanHostedHelper.getHostedAssetPath(coords, x)));
+  }
+
+  public String generateDownloadPackagesUrlsAsJson(
+      final ConanCoords coords,
+      final String repositoryUrl) throws JsonProcessingException
+  {
+    Map<String, String> downloadUrls = generateDownloadUrls(DOWNLOAD_URL_PACKAGE_ASSET_KINDS,
+        coords, repositoryUrl);
+    return ConanHostedHelper.MAPPER.writeValueAsString(downloadUrls);
+  }
+
+  public String generateDownloadUrlsAsJson(
+      final ConanCoords coords,
+      final String repositoryUrl)
+      throws JsonProcessingException
+  {
+    Map<String, String> downloadUrls = generateDownloadUrls(DOWNLOAD_URL_ASSET_KINDS, coords,
+        repositoryUrl);
+    return ConanHostedHelper.MAPPER.writeValueAsString(downloadUrls);
+  }
+
+  public String generatePackageSnapshotAsJson(final ConanCoords coords) throws JsonProcessingException {
+    Map<String, String> downloadUrls = DOWNLOAD_URL_PACKAGE_ASSET_KINDS
+        .stream()
+        .collect(
+            toMap(AssetKind::getFilename, x -> ConanHostedHelper.getHostedAssetPath(coords, x)));
+
+    Map<String, String> packageSnapshot = downloadUrls
+        .entrySet()
+        .stream()
+        .flatMap(entry -> {
+          String value = entry.getValue();
+          String hash = getHash(value);
+          if (hash != null) {
+            return Stream.of(Pair.of(entry.getKey(), hash));
+          }
+          return Stream.empty();
+        })
+        .collect(Collectors.toMap(Pair::getKey, Pair::getValue));
+    return ConanHostedHelper.MAPPER.writeValueAsString(packageSnapshot);
+  }
+
+  @Nullable
+  @TransactionalStoreBlob
```

Comment:
It seems like you don't store anything here. There is another transaction type

Reply:
fixed https://github.com/sonatype-nexus-community/nexus-repository-conan/pull/97/commits/4d00a4668a9352dc00162e2abe8ac600fda8696b